### PR TITLE
:tada: Add AB test for empty workspace set board tool by default

### PR DESCRIPTION
**Pull request by cocotime**
https://tree.taiga.io/project/penpot/us/9425?milestone=427670

### CODE REVIEW

File: frontend/src/app/main/data/workspace.cljs [hunk 1]
```diff
@@ -451,10 +452,13 @@
     ptk/WatchEvent
     (watch [_ state _]
       (if (dsh/lookup-page state page-id)
-        (let [file-id (:current-file-id state)]
+        (let [file-id (:current-file-id state)
+              objects        (deref refs/workspace-page-objects)
```
Comment (niwinz) [inline frontend/src/app/main/data/workspace.cljs]: Refs should almost never be used on events. Refs are derived atoms that points to state. But on events you already has the state.

In this case, you need `(dsh/lookup-page-objects state file-id page-id)`

File: frontend/src/app/main/data/workspace.cljs [hunk 2]
```diff
@@ -451,10 +452,13 @@
...
+              is-page-empty? (= (count (keys objects)) 1)]
```
Comment (niwinz) [inline frontend/src/app/main/data/workspace.cljs]: Yo can execute `count` on objects directly, without the need to access the `(keys objects)`

File: frontend/src/app/main/data/workspace.cljs [hunk 3]
```diff
@@ -451,10 +452,13 @@
...
           (rx/of (preload-data-uris page-id)
                  (dwth/watch-state-changes file-id page-id)
-                 (dwl/watch-component-changes)))
+                 (dwl/watch-component-changes)
+                 (when (and is-page-empty? (cf/external-feature-flag "boards-02" "test")) (dwd/select-for-drawing :frame))))
```
Comment (niwinz) [inline frontend/src/app/main/data/workspace.cljs]: I think that all this new code, can be encapsulated in a specific event and emit that event instead having all this logic executing all the time independently if the AB feature is activated or not.

**Review by niwinz — CHANGES_REQUESTED** [review]

File: frontend/src/app/main/data/workspace.cljs (hunk 1 above)
Comment (cocotime) [reply frontend/src/app/main/data/workspace.cljs]: fixed

File: frontend/src/app/main/data/workspace.cljs (hunk 2 above)
Comment (cocotime) [reply frontend/src/app/main/data/workspace.cljs]: fixed

File: frontend/src/app/main/data/workspace.cljs (hunk 3 above)
Comment (cocotime) [reply frontend/src/app/main/data/workspace.cljs]: I'm not sure if I did what you asked. I changed the code. Can you check if it's correct?

Comment (niwinz) [reply frontend/src/app/main/data/workspace.cljs]: Nice, yes, thanks

File: frontend/src/app/main/data/workspace.cljs [hunk 4]
```diff
@@ -273,6 +273,17 @@
         (rx/of (dws/select-shapes frames-id)
                dwz/zoom-to-selected-shape)))))
 
+(defn select-frame-tool
+  [page-id]
```
Comment (niwinz) [inline frontend/src/app/main/data/workspace.cljs]: Pass the file-id that you already have on the other event

File: frontend/src/app/main/data/workspace.cljs [hunk 5]
```diff
@@ -273,6 +273,17 @@
         (rx/of (dws/select-shapes frames-id)
                dwz/zoom-to-selected-shape)))))
 
+(defn select-frame-tool
```
Comment (niwinz) [inline frontend/src/app/main/data/workspace.cljs]: Use `defn-` for declare a private function

**Review by niwinz — CHANGES_REQUESTED** [review]

File: frontend/src/app/main/data/workspace.cljs (hunk 4 above)
Comment (cocotime) [reply frontend/src/app/main/data/workspace.cljs]: done

File: frontend/src/app/main/data/workspace.cljs (hunk 5 above)
Comment (cocotime) [reply frontend/src/app/main/data/workspace.cljs]: done

File: frontend/src/app/main/data/workspace.cljs [hunk 6]
```diff
@@ -273,6 +273,16 @@
         (rx/of (dws/select-shapes frames-id)
                dwz/zoom-to-selected-shape)))))
 
+(defn- select-frame-tool
+  [{:keys [page-id file-id]}]
```
Comment (niwinz) [inline frontend/src/app/main/data/workspace.cljs]: I think we don't need here a map for passing two static positional params. Avoid unnecesary allocation with a function with two arguments

```clojure
(defn- select-frame-tool
  [file-id page-id]
       ...
```

**Review by niwinz — CHANGES_REQUESTED** [review]
